options/ansi/generic/stdio-stubs.cpp

```diff
@@ -1045,7 +1045,9 @@ ssize_t getdelim(char **line, size_t *n, int delim, FILE *stream) {
 	}
 
 	char *buffer = *line;
-	size_t capacity = *n, nwritten = 0;
+	/* limit the starting capacity to 512 if buffer = NULL */
+	size_t capacity = (!buffer && *n > 512) ? 512 : *n;
```

Comment:
Should `*n` affect the buffer size at all if `!buffer`? What does the man page say? What do other libcs do?

Reply:
On `!buffer`, the man page tells us to ignore `*n`. glibc seems to set 120 if that is the case.